projects/js-packages/publicize-components/src/hooks/use-share-limits/index.ts

```diff
@@ -0,0 +1,120 @@
+import { useSelect } from '@wordpress/data';
+import { __, sprintf } from '@wordpress/i18n';
+import { store as socialStore } from '../../social-store';
+
+export type ShareLimits = {
+	status: 'approaching' | 'exceeded' | 'full' | 'none';
+	noticeType: 'default' | 'highlight' | 'warning' | 'error';
+	message: string;
+};
+
+/**
+ * Returns the messages for the share limits
+ *
+ * @param {number} shareLimit - Share limit
+ *
+ * @returns {ReturnType<typeof getMessages>} Share limits messages
+ */
+export function getMessages( shareLimit: number ) {
+	return {
+		default: sprintf(
+			// translators: %1$d is the number of shares allowed in 30 days.
+			__( 'Share limit for 30 days: %1$d.', 'jetpack' ),
+			shareLimit
+		),
+		exceeded: __(
+			'You have exceeded your share limit. Your posts will no longer be shared.',
+			'jetpack'
+		),
+		scheduled: __(
+			'You have scheduled posts that will not be shared because you will exceed the share limit.',
+			'jetpack'
+		),
+		approaching: __( 'You are approaching your share limit.', 'jetpack' ),
+	};
+}
+
+/**
+ * Returns the share limits details
+ *
+ * @returns {ShareLimits} Share limits details
+ */
+export function useShareLimits(): ShareLimits {
+	return useSelect( select => {
+		const store = select( socialStore );
+
+		const shareLimit = store.getShareLimit();
+		const totalSharesCount = store.getTotalSharesCount( { includeActiveConnections: true } );
+		const scheduledShares = store.getScheduledSharesCount();
+		const usedSharesCount = store.getSharesUsedCount();
+		const messages = getMessages( shareLimit );
+
+		let noticeType: ShareLimits[ 'noticeType' ] = 'default';
+		let status: ShareLimits[ 'status' ] = 'none';
+		let message = messages.default;
+
+		// If they have exceeded their limit
+		if ( totalSharesCount > shareLimit ) {
+			status = 'exceeded';
+			message = messages.exceeded;
+
+			/**
+			 * Here we have these cases:
+			 * 1. used >= limit: they have exceeded their limit without scheduled shares or active connections
+			 * 2. used < limit: they have exceeded their limit with scheduled shares or active connections
+			 * 2a. scheduled > 0: it means they have scheduled shares
+			 * 2b. scheduled = 0: it means they have active connections
+			 */
+			// Case 1
+			if ( usedSharesCount >= shareLimit ) {
+				noticeType = 'error';
+			}
+			// Case 2a
+			else if ( scheduledShares > 0 ) {
+				noticeType = 'warning';
+				message = messages.scheduled;
+			}
```

Comment:
This is all pretty complex isn't it? I can see what the switch statement was doing now, but I'm not sure one is much clearer than the other.

I think we could only be greater than the share limit with scheduled shares, and we would want to show an error at that point I think. 

The noticeType is usually the same for each message. `error => exceeded, warning => scheduled` etc. Perhaps the noticeType should be part of the messages data structure?

Having said this, I think perhaps we should wait before making further changes here until we change the design, we may end up showing completely different notices.

Reply:
Yeah, it's complicated